commit 600443618ad003148e754035ae9bfa0d17220ba3
Author: Konstantin Kolinko <kkolinko@apache.org>
Date:   Mon Nov 2 10:30:49 2009 +0000

    Mark JspCompilationContext#removed as volatile,
    because JspCompilationContext#isRemoved() is called from JspServletWrapper#service,
    and that access is not synchronized.
    
    git-svn-id: https://svn.apache.org/repos/asf/tomcat/trunk@831850 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/java/org/apache/jasper/JspCompilationContext.java b/java/org/apache/jasper/JspCompilationContext.java
index 5326c0e..bc5e948 100644
--- a/java/org/apache/jasper/JspCompilationContext.java
+++ b/java/org/apache/jasper/JspCompilationContext.java
@@ -81,7 +81,7 @@ public class JspCompilationContext {
 
     protected JspRuntimeContext rctxt;
 
-    protected int removed = 0;
+    protected volatile int removed = 0;
 
     protected URLClassLoader jspLoader;
     protected URL baseUrl;
